src/Lean/Meta/IndPredBelow.lean

```diff
@@ -574,7 +574,7 @@ def findBelowIdx (xs : Array Expr) (motive : Expr) : MetaM $ Option (Expr × Nat
 def mkBelow (declName : Name) : MetaM Unit := do
   if (← isInductivePredicate declName) then
     let x ← getConstInfoInduct declName
-    if x.isRec then
+    if x.isRec && !x.isNested then
```

Comment:
Could you please add a comment here that explains this condition?

Ideally this comment will make clear why we want to allow some inductives for which we can not generate `.below`.

It may be best to write this comment in the form of a doc-string on `mkBelow`, explaining when it is and isn't meant to work.

Reply:
I am not absolutely certain that it's not possible to generate `.below` for nested predicates, but the current implementation doesn't support them, and the modifications necessary to support them looks to be non-trivial. These nested predicates are still sound and accepted at a kernel level. As for why we want to allow them, some users have stumbled upon the error [in the past](https://proofassistants.stackexchange.com/questions/2045/unexpected-inductive-type-occurrence-error-for-recursive-inductive-type). Also, note that Lean already "accepts" nested predicates, in the sense that nested predicates still get silently added to the environment despite Lean not being able to generate their `.below` lemma, this fix simply avoids triggering the error in the first place. I'll nonetheless add documentation, and make a more complete/relevant test.

Reply:
Done.

Reply:
Back in the bug report issue could you:
1. Quote the actual error message that occurs, as a breadcrumb for the future.
2. Link to that `proofassistants` question as explanation of why this is needed.

Reply:
Sorry for the style problems, and thank you for spending time on this pull request. I've added the stack-exchange post and the error message in the issue, and fixed the code according to your suggestions.